PaymentHighway/UI/ExpiryDatePickerView.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  DateExpiryPicker.swift
+//  PaymentHighway
+//
+//  Created by Stefano Pironato on 25/09/2018.
+//  Copyright © 2018 Payment Highway Oy. All rights reserved.
+//
+
+import UIKit
+
+private let numberOfYears = 15
+
+class ExpiryDatePickerView: UIPickerView, UIPickerViewDelegate, UIPickerViewDataSource {
+    
+    private var months: [Int] = []
+    private var years: [Int] = []
+    private var currentMonth = Calendar.current.component(.month, from: Date())
+    
+    var onDateSelected: ((_ month: Int, _ year: Int) -> Void)?
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initialize()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initialize()
+    }
+    
+    override func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        notify()
+    }
+    
+    private func initialize() {
+        self.months = (1...12).map { $0 }
```

Comment:
Could omit explicit use of `self` here? There are a couple of similar cases in this file.

Reply:
agree not needed those self. removed! thanks